### PR TITLE
[Azure Search] Swagger changes to introduce new API version (2019-10-01-Preview)

### PR DESCRIPTION
**Pull request by tjacobhi**
### Description
This PR introduces a new API version (in preview) for the management surface of Azure Cognitive Search. This API version introduces the following APIs -

1. ListQueryKeys -> corrected from `get` in past API version to `post` (per RP guidance)
2. Update, Get and Delete private endpoint connection details -> Only these operations are managed via the search management API. "Creating" a private endpoint connection is initiated via NRP.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by tjacobhi** [comment]
Making Draft PR to handoff to team members.

**Review by arv100kri — DISMISSED** [review]

**Comment by arv100kri** [comment]
The only failing test is the "ignorable" prettier check (we'll fix this later)

**Review by jqhuangonearth — APPROVED** [review]

**Review by jqhuangonearth — APPROVED** [review]

**Comment by huangbolun** [comment]
@MyronFanQiu, @brjohnstmsft : please kindly review this change.. This is adding a new ARM management API version for Microsoft.Search to update/manage IP whitelisting feature and private endpoint support.

File: specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json [hunk 1]
```diff
@@ -0,0 +1,1465 @@
...
+          "x-ms-enum": {
+            "name": "EndpointAccess",
+            "modelAsString": false
+          }
+        },
+        "ipRules": {
+          "type": "array",
+          "description": "A list of rules that define inbound network access to the search service endpoint. By default, the search service endpoint is reachable from the internet.",
+          "items": {
+            "type": "object",
```
Comment (MyronFanQiu) [inline specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json]: just wonder the reason behind this structure? `items` only contains one field.

File: specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json [hunk 2]
```diff
@@ -0,0 +1,1465 @@
...
+          "SearchCheckNameAvailability": {
+            "$ref": "./examples/SearchCheckNameAvailability.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "checkNameAvailabilityInput",
+            "in": "body",
+            "description": "The resource name and type to check.",
+            "x-ms-client-flatten": true,
```
Comment (MyronFanQiu) [inline specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json]: Usually it appears in definition.

**Review by mmyyrroonn — APPROVED** [review]
LGTM. Wait for ARM's review

File: specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json (hunk 1 above)
Comment (huangbolun) [reply specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json]: refactored to model the item object in #/definitions/.

**Comment by huangbolun** [comment]
@MyronFanQiu , @brjohnstmsft , @anthony-c-martin : would you be able to merge the change? All checks have passed and the production is ready to go. Much appreciated.

**Pull request merged**